### PR TITLE
Add-join-Table

### PR DESCRIPTION
**Pull request by AJ193**
### Implementations in this project
- I created a table named vets with the following columns:
  id: integer (set it as autoincremented PRIMARY KEY)
  name: string
  age: integer
  date_of_graduation: date

- There is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a 
  species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.

- There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet 
  can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of 
  the date of the visit.

- I inserted the following data for vets:

  Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

- I inserted the following data for specialties:
  Vet William Tatcher is specialized in Pokemon.
  Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  Vet Jack Harkness is specialized in Digimon.

- I inserted the following data for visits:
  Agumon visited William Tatcher on May 24th, 2020.
  Agumon visited Stephanie Mendez on Jul 22th, 2020.
  Gabumon visited Jack Harkness on Feb 2nd, 2021.
  Pikachu visited Maisy Smith on Jan 5th, 2020.
  Pikachu visited Maisy Smith on Mar 8th, 2020.
  Pikachu visited Maisy Smith on May 14th, 2020.
  Devimon visited Stephanie Mendez on May 4th, 2021.
  Charmander visited Jack Harkness on Feb 24th, 2021.
  Plantmon visited Maisy Smith on Dec 21st, 2019.
  Plantmon visited William Tatcher on Aug 10th, 2020.
  Plantmon visited Maisy Smith on Apr 7th, 2021.
  Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  Angemon visited Jack Harkness on Oct 3rd, 2020.
  Angemon visited Jack Harkness on Nov 4th, 2020.
  Boarmon visited Maisy Smith on Jan 24th, 2019.
  Boarmon visited Maisy Smith on May 15th, 2019.
  Boarmon visited Maisy Smith on Feb 27th, 2020.
  Boarmon visited Maisy Smith on Aug 3rd, 2020.
  Blossom visited Stephanie Mendez on May 24th, 2020.
  Blossom visited William Tatcher on Jan 11th, 2021.

- I wrote queries to answer the following:
  Who was the last animal seen by William Tatcher?
  How many different animals did Stephanie Mendez see?
  List all vets and their specialties, including vets with no specialties.
  List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.

third 1
third 2
third 4

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -96,4 +96,58 @@ JOIN owners ON owners.id = animals.owner_id WHERE owners.full_name = 'Dean Winch
...
 SELECT O.full_name AS OWNER, COUNT(a.name) AS TOTAL_ANIMALS FROM owners AS O LEFT JOIN animals AS A
-ON A.owner_id = O.id GROUP BY O.full_name ORDER BY COUNT(A.name) DESC LIMIT 1; 
\ No newline at end of file
+ON A.owner_id = O.id GROUP BY O.full_name ORDER BY COUNT(A.name) DESC LIMIT 1;
+
+-- Who was the last animal seen by William Tatcher ?
+
+SELECT vets.name AS vets_name, animals.name AS animal_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id INNER JOIN animals ON animals.id = visits.animal_id WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC LIMIT 1;
```
Comment (mckent05) [inline queries.sql:105]: - You have done well so far, but the answer in the screenshot to for this query is wrong, please fix this. The last animal to visit William Tatcher should be `Blossom`. Please fix this 🙏. I suggest you carefully put in your data in the `data.sql`.

File: queries.sql [hunk 2]
```diff
@@ -96,4 +96,58 @@ JOIN owners ON owners.id = animals.owner_id WHERE owners.full_name = 'Dean Winch
...
+-- List all vets and their spcialties, including vets with no specialties.
+
+SELECT vets.name AS vets_name, species.name AS specialties FROM vets LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+SELECT vets.name AS vets_name, animals.name AS animal_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id INNER JOIN animals ON visits.animal_id = animals.id WHERE 
+vets.name = 'Stephanie Mendez' AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
```
Comment (mckent05) [inline queries.sql:121]: - Good job done so far, but the answer to this query in your screenshot is not correct, the animals that visited this vet within this date are, `Agumon` and `Blossom`

File: queries.sql [hunk 3]
```diff
@@ -96,4 +96,58 @@ JOIN owners ON owners.id = animals.owner_id WHERE owners.full_name = 'Dean Winch
...
+
+-- What animal has the most visits to vets?
+
+SELECT animals.name AS animal_name, COUNT(animals.id) AS most_visits FROM visits
+INNER JOIN animals ON visits.animal_id = animals.id GROUP BY animals.name ORDER BY COUNT(animals.id) DESC LIMIT 1;
```
Comment (mckent05) [inline queries.sql:126]: - Good job done so far, but the answer to this query in your screenshot is not correct, the animal that has the most visit to the vet is `Boarmon`. Please correct this 😄

File: queries.sql [hunk 4]
```diff
@@ -96,4 +96,58 @@ JOIN owners ON owners.id = animals.owner_id WHERE owners.full_name = 'Dean Winch
...
+
+-- Who was Maisy Smith's first visit?
+
+SELECT vets.name AS vets_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id WHERE vets.name = 'Maisy Smith' ORDER BY visits.date_of_visit
+LIMIT 1;
```
Comment (mckent05) [inline queries.sql]: - Good job done so far, please include the name of the animal that first visited `Maisy Smith` not just the date. Please fix this 🙏

File: queries.sql [hunk 5]
```diff
@@ -96,4 +96,58 @@ JOIN owners ON owners.id = animals.owner_id WHERE owners.full_name = 'Dean Winch
...
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+
+SELECT animals.name AS animal_name, animals.date_of_birth AS date_of_birth, animals.escape_attempts AS escape_attempts, 
+animals.neutered AS neutered, animals.weight_kg AS weight_kg, species.name AS species_type, vets.name AS vets_name, 
+vets.age AS vets_age, vets.date_of_graduation AS date_of_graduation, visits.date_of_visit AS date_of_visit FROM visits 
+INNER JOIN animals ON visits.animal_id = animals.id INNER JOIN species ON animals.species_id = species.id
+INNER JOIN vets ON visits.vet_id = vets.id ORDER BY visits.date_of_visit DESC LIMIT 1;
```
Comment (mckent05) [inline queries.sql:142]: - Good job done so far, but the answer to this query is wrong, the animal who visited most recently is `Devimon`, every other info is correct except the animal's name. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @AJ193, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job add a join table for visits to your database 👏 

## Required Changes ♻️
_Check the comments under the review._
- You have done well so far, but you seem to be having wrong results for some of your queries because you have inputted the wrong details in your `data.sql` file. I suggest you carefully input the data again and update the screenshots for the right results in your PR. Please fix this 😄 

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @AJ193, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Correct schema ✔️ 
- Correct data ✔️ 
- Correct queries ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**